src/store/reader.rs

```diff
@@ -329,6 +294,31 @@ impl StoreReader {
     }
 }
 
+fn block_read_index(block: &[u8], doc_pos: u32) -> crate::Result<Range<usize>> {
+    let doc_pos = doc_pos as usize;
+    let size_of_u32 = std::mem::size_of::<u32>();
+
+    let index_len_pos = block.len() - size_of_u32;
+    let index_len = u32::deserialize(&mut &block[index_len_pos..])? as usize;
+
+    if doc_pos > index_len {
+        return Err(crate::TantivyError::InternalError(
+            "Attempted to read doc from wrong block".to_owned(),
+        ));
+    }
+
+    let index_start = block.len() - (index_len + 1) * size_of_u32;
+    let index = &block[index_start..index_start + index_len * size_of_u32];
+
+    let start_pos = u32::deserialize(&mut &index[doc_pos * size_of_u32..])? as usize;
```

Comment:
```suggestion
    let start_offset = u32::deserialize(&mut &index[doc_pos * size_of_u32..])? as usize;
```
_offset is imo better to name byte offsets